draft-irtf-cfrg-ristretto255-decaf448.md

```diff
@@ -748,24 +726,21 @@ w3 = v_prime * s * (r - 1) * ONE_MINUS_TWO_D + sgn
 ## Scalar field
 
 The scalars for the decaf448 group are integers modulo the order l
-of the decaf448 group.
+of the decaf448 group. Note that this is the same scalar field as
+edwards448, allowing existing implementations to be reused.
 
-Scalars are encoded as bytestrings of 56 bytes in length, in little-endian order.
+Scalars are encoded as 56-byte strings in little-endian order.
 Implementations **SHOULD** check that any scalar s falls in the range
 0 <= s < l when parsing them and reject non-canonical scalar
 encodings. Implementations **SHOULD** reduce scalars modulo l when
-encoding them as bytestrings.
+encoding them as byte strings. Omitting these strict range checks is
+**NOT RECOMMENDED** but is allowed to enable reuse of scalar
+arithmetic implementations in existing edwards448 libraries.
 
-Given a uniformly distributed bytestring b of 64 bytes length, implementations can
-obtain a scalar by interpreting b as a 512-bit
+Given a uniformly distributed 64-byte string b, implementations can
+obtain a scalar by interpreting the 64-byte string as a 512-bit
```

Comment:
Hmm, does this claim actually hold? The [analysis in hash-to-curve](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-14#section-5.1) would suggest that you'd need at least 72 bytes to reduce bias to 2^-128. This seems to reduce bias to 2^-64.

Reply:
`hash_to_field` suffers from being general, in that it tries to apply for any `p`. The bias depends heavily on how far `p` (or, here, `l`) is from a power of two. Here we know the exact value, so we can analyze the bias precisely.

See https://github.com/cfrg/draft-irtf-cfrg-ristretto255/pull/9#issuecomment-985779317 for Mike's analysis, and note how 512 is actually overkill, although convenient in that it's consistent with r255.

Reply:
Ah, cool, thanks for educating me!